File: 2015/2/typescript/main.ts
import { readFileSync, existsSync, PathLike } from "fs";

function main() {
  // get command line arguments
  const args: String[] = process.argv.slice(1);

  if (args.length < 2) {
    console.log("Build: tsc main.ts");
    console.log("Usage: main main.js <input_file>");
    return;
  }

  const input_file: String = args[1];

  if (!existsSync(input_file as PathLike)) {
    console.log("File not found: ", input_file);
    return;
  }

  const input = readFileSync(input_file as PathLike, "utf8");

  const lines: String[] = input.split("\n");

  let total = 0;
  for (let line of lines) {
    // split line by x
    const parts: String[] = line.split("x");

    if (parts.length != 3) {
      console.log("Invalid input: ", line);
      continue;
    }

    const l: number = parseInt(parts[0] as string);
    const w: number = parseInt(parts[1] as string);
    const h: number = parseInt(parts[2] as string);

    const sides: number[] = [l * w, w * h, h * l];
    const min_side: number = Math.min(...sides);

    total += 2 * sides.reduce((a, b) => a + b) + min_side;
  }

  console.log(total);
}

main();

File: 2015/2/python/main.py
#!/usr/bin/env python3

import sys
import os

def main():
    args = sys.argv[1:]

    if len(args) == 0:
        print("Usage: python3 main.py <input_file>")
        return
    
    input_file = args[0]
    if not os.path.exists(input_file):
        print("File not found")
        return
    
    lines = []
    with open(input_file, "r") as f:
        lines = f.readlines()
    
    total = 0
    for line in lines:
        line = line.split('x')
        
        l = int(line[0])
        w = int(line[1])
        h = int(line[2])
        
        area = 2*l*w + 2*w*h + 2*h*l
        min_side = min(l*w, w*h, h*l)
        total += area + min_side
    
    print(total)



if __name__ == "__main__":
    main()